cpp/include/cugraph_c/sampling_algorithms.h

```diff
@@ -319,8 +319,59 @@ void cugraph_sampling_set_dedupe_sources(cugraph_sampling_options_t* options, bo
  */
 void cugraph_sampling_options_free(cugraph_sampling_options_t* options);
 
+/**
+ * @brief     Opaque neighborhood sampling heterogeneous fanout type
+ */
+// FIXME: internal representation should be tuple instead of pairs - Make it more generic (tuple)
```

Comment:
For case 2, since the user is only passing offsets and fan-out values. How would we figure out the edge types since they will only be passing the offsets they care about.